Added support for oauth2 config

mcalster opened the pull request:
<!--- 

**IMPORTANT:** Please submit pull requests to [alexkappa/terraform-provider-auth0](https://github.com/alexkappa/terraform-provider-auth0). This helps maintainers organize work more efficiently.

See what makes a good Pull Request at : https://github.com/alexkappa/terraform-provider-auth0/blob/master/.github/CONTRIBUTING.md#pull-requests 

--->
### Proposed Changes

Fixes #266

#### Acceptance Test Output

```
$ make testacc TESTS=TestAccConnectionOAuth2
==> Checking that code complies with gofmt requirements...
?       github.com/alexkappa/terraform-provider-auth0   [no test files]
=== RUN   TestAccConnectionOAuth2
--- PASS: TestAccConnectionOAuth2 (2.87s)
PASS
coverage: 12.5% of statements
ok      github.com/alexkappa/terraform-provider-auth0/auth0     3.258s  coverage: 12.5% of statements
?       github.com/alexkappa/terraform-provider-auth0/auth0/internal/debug     [no test files]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok      github.com/alexkappa/terraform-provider-auth0/auth0/internal/random    0.333s                                                                                                     coverage: 0.0% of statements [no tests to run]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok      github.com/alexkappa/terraform-provider-auth0/auth0/internal/validation                                                                                                           0.105s   coverage: 0.0% of statements [no tests to run]
?       github.com/alexkappa/terraform-provider-auth0/version   [no test files]
```

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

<!--- Thank you for keeping this note for the community --->

mcalster commented:
This is blocked blocked by #https://github.com/go-auth0/auth0/pull/140

mcalster commented:
@alexkappa I think this getting close to done. Can you have a look?

mcalster commented:
@alexkappa this should now also be ready :-)

alexkappa commented:
Hi @mcalster, I've tagged your changes in go-auth0 to [v0.4.6](https://github.com/go-auth0/auth0/tree/v4.6.0). If we update go.mod we are good to merge this ;)

mcalster commented:
Awesome 👍  @alexkappa I've updated the version and updated the test results.

mcalster commented:
> Awesome 👍 @alexkappa I've updated the version and updated the test results.



> Hi @mcalster, I've tagged your changes in go-auth0 to [v0.4.6](https://github.com/go-auth0/auth0/tree/v4.6.0). If we update go.mod we are good to merge this ;)

When do you think you can get it merged? I'd like to use it this week if possible! 🥇

alexkappa commented:
@mcalster released a new version which includes your changes, many thanks!

mcalster commented:
Thanks! I'm already setting up to use it 🙂


tor. 24. sep. 2020 kl. 18.12 skrev Alex Kalyvitis <notifications@github.com
>:

>
>
> @mcalster <https://github.com/mcalster> released a new version which
> includes your changes, many thanks!
>
>
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/alexkappa/terraform-provider-auth0/pull/267#issuecomment-698443066>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAX7IIZASLPOGNU4M5SNCOTSHNVWHANCNFSM4RPPZJTQ>
> .
>
>
> --
-- Venlig Hilsen
Kim Alster
